Makefile

```
@@ -288,6 +296,60 @@ test_no_lint:
 
 test: lint test_no_lint
 
+_build_container_image:
+	@[ -z "$$BASE_IMAGE" ] && { echo "BASE_IMAGE must be set"; exit 1; } || \
+	[ -z "$$TEST_IMAGE" ] && { echo "TEST_IMAGE must be set"; exit 1; }; \
+	podman image exists "$$TEST_IMAGE" && exit 0; \
+	echo "=========== Preparing container test env image ==========="; \
+	_CONT_NAME="$${TEST_IMAGE}-cont"; \
+	podman run -di --name $$_CONT_NAME -v "$$PWD":/repo:Z $$BASE_IMAGE && \
+	podman exec $$_CONT_NAME dnf -y install make && \
+	podman exec $$_CONT_NAME rm -rf repocopy && \
+	podman exec $$_CONT_NAME mkdir /repocopy && \
+	podman exec $$_CONT_NAME /bin/bash -c "cp -a /repo/* /repocopy" && \
+	podman exec -w /repocopy -e PYTHON_VENV=$$_VENV $$_CONT_NAME make install-deps-fedora && \
+	podman commit $$_CONT_NAME $$TEST_IMAGE && \
+	podman kill $$_CONT_NAME; \
+	podman rm $$_CONT_NAME
+
+test_container:
+	@case $(_TEST_CONTAINER) in \
+	f34) \
+		export BASE_IMAGE="fedora:34"; \
+		export TEST_IMAGE="leapp-repo-tests-f34"; \
+		export _VENV="python3.9"; \
+		;; \
+	rhel8) \
+		export BASE_IMAGE="registry.access.redhat.com/ubi8/ubi:latest"; \
+		export TEST_IMAGE="leapp-repo-tests-rhel8"; \
+		export _VENV="python3.6"; \
+		;; \
+	*) \
+		echo "Error: Available containers are: f34, rhel8"; exit 1; \
+		;; \
+	esac; \
+	$(MAKE) _build_container_image && \
+	echo "=========== Running tests in $(_TEST_CONTAINER) container ===============" && \
+	_CONT_NAME="leapp-repo-run-tests-$(_TEST_CONTAINER)-cont"; \
+	podman run -di --name $$_CONT_NAME -v "$$PWD":/repo:Z $$TEST_IMAGE && \
+	podman exec $$_CONT_NAME rm -rf /repocopy/repos && \
+	podman exec $$_CONT_NAME cp -a /repo/repos /repocopy && \
+	podman exec -e PYTHON_VENV=$$_VENV -w /repocopy $$_CONT_NAME make clean && \
+	podman exec -e PYTHON_VENV=$$_VENV -w /repocopy $$_CONT_NAME make $${_TEST_CONT_TARGET:-test}; \
+	podman kill $$_CONT_NAME; \
+	podman rm $$_CONT_NAME
+
+test_no_lint_container:
+	_TEST_CONT_TARGET="test_no_lint" $(MAKE) test_container
+
+test_container_all:
+	for container in "f34" "rhel8"; do \
+		TEST_CONTAINER=$$container $(MAKE) test_container; \
+	done
+
+clean_containers:
+	podman rmi -i "leapp-repo-tests-f34" "leapp-repo-tests-rhel8"
```

To be compatible with older versions of podman
```suggestion
	for i in "leapp-repo-tests-f34" "leapp-repo-tests-rhel8"; do \
               podman rmi -f $$i >/dev/null 2>&1 || : ; \
       done

```